src/Agent.Worker/TaskManager.cs

```diff
@@ -69,11 +69,11 @@ public sealed class TaskManager : AgentService, ITaskManager
             {
                 try
                 {
-                    //sometimes the temp folder is not deleted -> wipe it
+                    //if the temp folder wasn't moved -> wipe it
                     if (Directory.Exists(tempDirectory))
                     {
                         Trace.Verbose("Deleting task temp folder: {0}", tempDirectory);
-                        Directory.Delete(tempDirectory, true);
+                        IOUtil.DeleteDirectory(tempDirectory);
```

Comment:
Based on some quick testing on my machine, it *looks like* zip files can contain readonly files. I didn't send the file over the web and back, but it looks like zip understands the concept of readonly files.